pkg/planner/util/path.go

```diff
@@ -51,6 +51,12 @@ type AccessPath struct {
 	// PartialIndexPaths store all index access paths.
 	// If there are extra filters, store them in TableFilters.
 	PartialIndexPaths []*AccessPath
+	// For every dnf/cnf item, there maybe several matched partial index paths to be determined later in property detecting and cost model.
+	PartialAlternativeIndexPaths [][]*AccessPath
+	ShouldBeKeptCurrentFilter    bool
+	// ShouldBeKeptCurrentFilter indicates that there are some expression inside this dnf that couldn't be pushed down, and we should keep the entire dnf above.
+	ShouldBeKeptCurrentFilterExpression expression.Expression
```

Comment:
Some suggestions about the comments and naming:
1. Make it clear that an access path with `PartialAlternativeIndexPaths` is a different one, and it can't have `PartialIndexPaths`, and a single index or table path also doesn't use this field.
2. Specify that `ShouldBeKeptCurrentFilter` and `ShouldBeKeptCurrentFilterExpression` are only used with `PartialAlternativeIndexPaths`, which means for the new type of access path.
3. "should be kept" sounds a bit strange to me, and so does the `CurrentFilter`. How about changing the names, like `IndexMergeSourceFilter` and `KeepIndexMergeSourceFilter`?